### PR TITLE
[Bug] Fix planar shadow with instancing

### PR DESCRIPTION
**Pull request by troublemaker52025**
Re: #

### Changelog

*

-------

### Continuous Integration

This pull request:

* [x] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Comment by troublemaker52025** [comment]
@cocos-robot run test cases

**Comment by troublemaker52025** [comment]
* https://github.com/cocos/cocos-engine/issues/13255

**Review by jk20012001 — APPROVED** [review]

File: native/cocos/scene/Model.cpp [hunk 1]
```diff
@@ -167,6 +167,10 @@ void Model::updateUBOs(uint32_t stamp) {
 
     updateSHUBOs();
 
+    const auto *pipeline = Root::getInstance()->getPipeline();
+    const auto *shadowInfo = pipeline->getPipelineSceneData()->getShadows();
+    const auto forceUpdateUBO = shadowInfo->isEnabled() && shadowInfo->getType() == ShadowType::PLANAR;
```
Comment (minggo) [inline native/cocos/scene/Model.cpp:172]: Can `shadowInfo ` be nullptr?

Comment (troublemaker52025) [reply native/cocos/scene/Model.cpp:172]: If pipelineSceneData exists, it will not be destroyed

**Review by minggo — APPROVED** [review]

**Pull request merged**